### ISSUE TITLE
Support extended Hugo version

### ISSUE THREAD

**Issue by DirtyF** [netlify/build-image#182]
Duplicate of: https://github.com/netlify/binrc/issues/16
Opening an issue here as @futuregerald kindly suggested.

**- What is the current behavior?**

From version 0.43 Hugo comes in two flavours, extended version allows native Sass compilation.
Netlify currently does not support extended version:

```
2:28:30 PM: ERROR 2018/07/10 12:28:30 error: failed to transform resource: failed to transform “scss/main.scss” (text/x-scss): this feature is not available in your current Hugo version
```

**Edit: For now, the recommended workflow is to commit your `/resources` folder after building locally with `hugo`.**

**- What is the expected behavior?**

Either download the extended binary by default from now on or allow to specify something like 0.43/extended in netlify.toml.

_extended is added to the release path before the version number:

https://github.com/gohugoio/hugo/releases/download/v0.43/hugo_0.43_Linux-64bit.tar.gz
https://github.com/gohugoio/hugo/releases/download/v0.43/hugo_extended_0.43_Linux-64bit.tar.gz

**Comment by jensamunch** [comment]
i tried manually including binary in the repo - as a quick-fix - doesn't work:

```
6:20:18 PM: Executing user command: ./hugo --config paceup.com.toml,config.toml
6:20:18 PM: ./hugo: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by ./hugo)
6:20:18 PM: ./hugo: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by ./hugo)
```

**Comment by fool** [comment]
We'll track that separate issue here, @jensamunch : https://github.com/netlify/build-image/issues/183

**Comment by baublet** [comment]
To those looking for a workaround until this is fixed:

Run `hugo` locally with the extended version, then check in your `resources/_gen/assets` directory to the repository so that Hugo skips trying to generate the assets.

**Comment by jensamunch** [comment]
It would be really great to have a bit more of an explanation from the Netlify team on timeline and plan.

**Comment by brycekahle** [comment]
@jensamunch Unfortunately, the solution involves upgrading the version of Ubuntu we use in the base image. That is a larger user impact than we initial hoped for. In order to ensure a smooth transition for all users, we need to do some work around communicating the changes ahead of time. Once we have a firm timeline, we will communicate that publicly.

**Comment by igk1972** [comment]
Fast workaround https://github.com/netlify/build-image/issues/183#issuecomment-419199649

**Comment by rdebeasi** [comment]
If you don't want to check in `resources` or install packages on build, here are a couple of other workarounds:

- Instead of sass pipelines, use [victor-hugo](https://github.com/netlify-templates/victor-hugo) plus [postcss-sass](https://github.com/jonathantneal/postcss-sass).
- Instead of image pipelines, use ImageMagick. [Here's an example](https://github.com/openpracticelibrary/openpracticelibrary/blob/staging/build.sh#L54) of how we're using Imagemagick in our Netlify builds.

**Comment by bep** [comment]
@rdebeasi you don't need the "extended version" to do image processing. 

I was told by Netlify several times that this issue is hanging on a upgrade of their build environments from Ubuntu 14.04 (which has an EOL date in April next year).

I have been given the impression that this is something "actively worked on" and that solution was "around the corner". I would probably been looking for an alternative fix from the Hugo side (building 14.04 compatible binaries) if I had known that this was going to take so long.

Is it possible to get a time estimate on this issue?

**Comment by eberkund** [comment]
For what it's worth the solution posted by @igk1972 is working great for me

**Comment by bcomnes** [comment]
Hey all, I'm taking over for Bryce on this issue.  Sounds like this is blocked by https://github.com/netlify/build-image/pull/222

The Ubuntu upgrade is in progress (see the mentioned PR), and I am currently working on improving our build pipeline to accommodate different build images, and free up constraints to help make more frequent releases.  I can't give dates yet, but it's high priority and I am actively working on it.

**Comment by sijad** [comment]
the error gone by adding `dist: xenial` to my `.travis.yml`, see https://blog.travis-ci.com/2018-11-08-xenial-release

**Comment by bwklein** [comment]
Looks like #222 was merged to dev, but I don't know what happens after that.

**Comment by bep** [comment]
>is this now solved?

@jensamunch No, but @biilmann (Netlify CEO) has silently kind of at least not denied that a fix for this could be a nice Christmas present :-)

**Comment by bcomnes** [comment]
Hi All!  

Any volunteers interested in beta testing a new buildbot image?  

What it entails:

- Provide me with the netlify slug of the site you would like tested on the new image.
- We will be testing https://github.com/netlify/build-image/issues/183#issuecomment-419199649
- I will flag the site to use the beta image until we land a fix or you request to be taken back to the standard image.

Im particularly interested in sites that may be running not in extended mode but on a version of hugo greater than 0.43.

**Comment by bwklein** [comment]
@bcomnes 'tecidev'

**Comment by bcomnes** [comment]
> 'tecidev'

@bwklein enabled

**Comment by romant** [comment]
@bcomnes - `romant`

thank you !

**Comment by bwklein** [comment]
@bcomnes It works as expected. Thank you!

**Comment by davidsneighbour** [comment]
Is it possible to enable for "free" sites? Then please eloquent-morse-196fd2

**Comment by nguyenvanduocit** [comment]
@bcomnes Please add me to the test: "12bit-blog"

**Comment by jamescrowley** [comment]
@bcomnes please add kind-bartik-cd167d too please, thanks!

**Comment by StefanoChiodino** [comment]
@bcomnes please add "negative-reviews", cheers!

**Comment by pavelbinar** [comment]
@bcomnes `remoteorigin`.
Thank you.

**Comment by bcomnes** [comment]
Hey folks, sorry for the delay.  I was out for holidays and a handoff wasn't possible.  

I deployed https://github.com/netlify/build-image/pull/197 yesterday, but determined it wasn't a reliable enough fix.  I partially reverted it here: https://github.com/netlify/build-image/pull/248

I kept the required libraries though, which means if you provide your own hugo extended binary in the build, or download it on build, you can use it by doing something like:

```console
LD_LIBRARY_PATH=$HOME/stdc++6/usr/lib/x86_64-linux-gnu ./hugo
```

The good news is, I've started a deployment of the `dev` branch which is running ubuntu 16 and supports hugo extended by default.  Once that's ready for testing, I will set all of the above requests to use that image instead.  Soon after, we'll make switching between the `precise` image and the `dev` image will be user serviceable in the site settings. 

Stay posted on this issue for updates, and thank you for your patients on this issue.  It is my primary priority as of now.

**Comment by StefanoChiodino** [comment]
@bcomnes seems broken for me:

    6:29:46 AM: hugo: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by hugo)

If you could please revert it, I need to set something live.

Cheers

**Comment by rgiese** [comment]
@bcomnes if you're still taking signups, please enable for 'grumpycorp'. Thank you!

**Comment by bcomnes** [comment]
Sorry for the delay.  Almost ready to start adding sites to the dev image this week.  Please update your comments if you do NOT want your site added to a potentially-unstable-at-times `dev` image beta.

**Comment by y0ns0n** [comment]
@bcomnes please add 'yonson', thanks!

**Comment by bcomnes** [comment]
Ok, confirmed its working, I'm going to start adding sites onto it.

**Comment by bcomnes** [comment]
Ok, I set up an issue track dev image beta stuff over at https://github.com/netlify/build-image/issues/254

Lets keep this issue specific to hugo extended from here on out.

**Comment by bcomnes** [comment]
Since this is working in the xenial image, I'm going to close this.  We should get that out to everyone soonish.  Thanks for your patients and all the input on this one, and my personal apologies for it taking so long.  We have some great improvements to the buildbot pipeline coming this year you will all love.

**Comment by mischkl** [comment]
Hi, just a question, how do I enable the extended version of Hugo on my site now that it's available to everyone?

**Comment by junedev** [comment]
If other people like me come here wondering how to actually apply the fix to their existing projects, here is the article how to switch to the new Ubuntu version: https://www.netlify.com/blog/2019/03/14/a-more-flexible-build-architecture-with-updated-linux/
After that you can safely include the resources/_gen folder in your .gitignore file (and remove them from git tracking).

**Comment by beausmith** [comment]
Solution in https://github.com/netlify/build-image/issues/183#issuecomment-554689821 allowed me to deploy hugo with sass/scss on netlify.
